cheroot/server.py

```diff
@@ -156,7 +156,7 @@
 ASTERISK = b'*'
 FORWARD_SLASH = b'/'
 QUOTED_SLASH = b'%2F'
-QUOTED_SLASH_REGEX = re.compile(b'(?i)' + QUOTED_SLASH)
+QUOTED_SLASH_REGEX = re.compile(b''.join([b'(?i)', QUOTED_SLASH]))
```

Comment:
Let's use format-interpolation here.

Reply:
I had an issue with that one in particular.  Format interpolation does not work with bytes.  I am probably missing something silly here.  

```
>>> x = b'(?i)' + b'%2F'
>>> print(x)
b'(?i)%2F'
>>> y = b'{}{}'.format(b'(?i)', b'%2F')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'bytes' object has no attribute 'format'
>>> y = '{}{}'.format(b'(?i)', b'%2F')
>>> print(y)
b'(?i)'b'%2F'
```

I suppose you could use another line.

```
QUOTED_SLASH_STRING = '%2F'
```

It did not seem like the correct solution though.

Reply:
In such a case, maybe let's just skip these cases. We could create a flakehell legacy baseline that would ignore these for now... And when we drop Python 2 support in 2020 the problem will disappear.

Reply:
Just checked that bytes don't have format in Python 3.8 too.

Reply:
The only change I'll do is turn the join arg into a tuple